src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java

```diff
@@ -306,17 +310,22 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
             writeToLog(String.format("Using App '%s'", env.expand(appArtifact)));
             Upload appUpload = adf.uploadApp(project, appArtifact);
             String appArn = appUpload.getArn();
-            String appName = String.format("%s (Jenkins)", appUpload.getName());
-
+            String deviceFarmRunName = null;
+            if (StringUtils.isBlank(runName)) {
+            	deviceFarmRunName = String.format("%s", env.get("BUILD_TAG"));
```

Comment:
Why do you just set the default value on the text field (in the Jelly file) to contain the variable string for the BUILD_TAG?

That would simplify this case as you don't have to handle it in code at all.